src/Npgsql/Internal/ResolverFactories/JsonDynamicTypeInfoResolverFactory.cs

```diff
@@ -53,6 +53,14 @@ JsonSerializerOptions SerializerOptions
         readonly Type[] _jsonClrTypes = jsonClrTypes ?? [];
         TypeInfoMappingCollection? _mappings;
 
+#if NET9_0_OR_GREATER
```

Comment:
Did you mean to multi-target for `net9.0`? As of right now, this code is dead since this project doesn't have a `net9.0` targets.

Reply:
Yep, we're aware - this is just to make sure we get the proper thing if and when we do target net9.0+ (and won't use reflection...).